### DISCUSSION TITLE
Question: Support for model with app specific gui (DOM) and screenshot functionality

### DISCUSSION THREAD

**Discussion by tom-r-o** [google/model-viewer#1949] [Q&A]
### Description

My requirement - show an AR model with:
- Application specific ui (add a couple of playback buttons)
- Support taking a screenshot

IIUC the webxr mode does not allow taking a screenshot, but Scene View does.
Is is possible to add DOM elements on top of Scene Viewer?

Note: when searching "panda" in google in an android device there's a Scene View mode that has specialized UI. Is this API available and documented?

### Browser Affected

- [x] Chrome
- [ ] Edge
- [ ] Firefox
- [ ] IE
- [x] Safari

### OS

- [x] Android
- [x] iOS
- [ ] Linux
- [ ] MacOS
- [ ] Windows

**Comment by elalish** [comment]
No, DOM customization not part of scene-viewer's API, and is unlikely to be since it is an app external to the browser. However, we are working to bring screenshots to WebXR.

**Comment by tom-r-o** [comment]
To clarify, when googling "panda" and entering the AR mode there's a gallery with Scene View - how is it made? (is it an android application?)

![image](https://user-images.githubusercontent.com/1384382/103355071-0139ca80-4ab6-11eb-8145-5e88968951d9.png)

**Comment by elalish** [comment]
Yes, scene-viewer is an Android application, as is AGSA (the Android Google Search App, which scene-viewer is a part of).
